pkg/vm/client.go

```diff
@@ -86,22 +92,22 @@ func (c *Client) Resume(ctx context.Context) error {
 }
 
 // Inspect return information about the vm
-func (c *Client) Inspect(ctx context.Context) (CPU, MemMib, error) {
+func (c *Client) Inspect(ctx context.Context) (VMData, error) {
 	request, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://unix/api/v1/vm.info", nil)
 	if err != nil {
-		return 0, 0, err
+		return VMData{}, err
 	}
 	request.Header.Add("content-type", "application/json")
 
 	response, err := c.client.Do(request)
 	if err != nil {
-		return 0, 0, errors.Wrap(err, "error calling machine info")
+		return VMData{}, errors.Wrap(err, "error calling machine info")
 	}
 
 	defer response.Body.Close()
 
 	if response.StatusCode != http.StatusOK {
```

Comment:
is it pissible to have a body that we can also add in the error message along with the status code?